torch/csrc/utils/disable_torch_function.cpp

```diff
@@ -227,6 +272,24 @@ auto check_has_torch_function(PyObject* obj, bool ignore_mode) -> bool {
       !THPVariable_CheckTypeExact(tp) && !is_basic_python_type(tp) &&
       torch::torch_function_enabled() && has_torch_function_attr(obj));
 }
+
+bool has_torch_function(PyObject* obj) {
+  return (
+      !torch::should_skip_torch_function() &&
+      torch::check_has_torch_function(obj));
+}
```

Comment:
This seems a bit dangerous: this predicate is side-effecting, but it is non-obvious from the name. Easy to accidentally call it twice (thinking that it's idempotent), but actually the next time around it will report true. This may be the source of errors.

Reply:
Is the name the main issue, or would you prefer that the caller explicitly checks `should_skip_torch_function`?

Reply:
It's related to https://github.com/pytorch/pytorch/pull/74728#issuecomment-1208446254 . Essentially, it is non-obvious that this function call is what causes the skip boolean to toggle. If it's done via an RAII guard, it's a lot clearer that there is some change (which will get reverted upon exiting the scope, rather than potentially escaping to unbounded scope).